### ISSUE TITLE
Error: Incompatible wire encryption levels requested on client and server    at doCallback

### ISSUE THREAD

**Issue by khalidMhd** [hgourvest/node-firebird#317]

**Comment by khalidMhd** [comment]
I am facing these two error
Error: Incompatible wire encryption levels requested on client and server at doCallback
Error2: Incompatible wire encryption levels requested on client and server

+++++++++++++ code review ++++++++++++++

const firebird = require('node-firebird');

const options = {
host: '127.0.0.1',
port: 3050,
database: 'C:/Users/khalid.d/Documents/MYDB.fdb',
user: 'SYSDBA',
password: 'root',
lowercase_keys: false,
pageSize: 4096,
role: null,
// charset: 'WIN1253',
// encrypt: 'AES256',
// algorithm: 'AES256',
};

exports.executeQuery = async (query, params = []) => {
return new Promise((resolve, reject) => {
firebird.attach(options, (err, db) => {
if (err) {
console.error('Error connecting to the Firebird database:', err);
reject(err);
return;
}
console.log('Connected to the Firebird database.');

  db.query(query, params, (err, result) => {
    if (err) {
      console.error('Error executing query:', err);
      reject(err);
    } else {
      console.log('Query executed successfully.');
      resolve(result);
    }

    db.detach();
  });
});
});
};

**Comment by maratth** [comment]
Duplicate #318

**Comment by mariuz** [comment]
"Incompatible wire encryption levels requested on client and server" is the server rejecting the crypt negotiation. Since this was filed, the driver gained full wire encryption (Arc4 and ChaCha/ChaCha64, negotiated automatically with Srp session keys) — current 2.x versions attach cleanly against servers with `WireCrypt = Enabled` or `Required` (our CI runs against `Enabled` on Firebird 3–6).

Today this error only appears on a genuine configuration mismatch:

- server `WireCrypt = Required` + client `wireCrypt: Firebird.WIRE_CRYPT_DISABLE` → remove the disable (encryption is on by default), or
- server `WireCrypt = Disabled` + a client requiring encryption → align the server setting, or
- **Legacy_Auth**: the legacy plugin has no session key, so encryption is impossible — a `Required` server refuses such connections by design; use Srp.

Closing as resolved by the shipped wire-crypt support — reopen with driver + server versions and both settings if a current 2.x still fails outside that matrix.
